src/app/common/money/fiat-conversion.ts

```diff
@@ -0,0 +1,18 @@
+import type { CryptoCurrencies } from '@shared/models/currencies.model';
+import { createMarketData, createMarketPair } from '@shared/models/market.model';
+import { type Money } from '@shared/models/money.model';
+
+import { baseCurrencyAmountInQuote } from './calculate-money';
+
+export function checkIsMoneyAmountGreaterThanZero(money: Money) {
+  return !(money.amount.isNaN() || money.amount.isZero());
+}
+
+export function convertCryptoCurrencyMoneyToFiat(
+  currency: CryptoCurrencies,
+  price: Money,
```

Comment:
Shouldn't we use `MarketData` type here?

Reply:
The `createMarketData` is used inside this function, so you are saying here better to do outside and pass in? Not sure the function is really needed then, was trying to create a shared function to `createMarketData` and then return `baseCurrencyAmountInQuote` ...but can just repeat the code where needed?

Reply:
I idea is that, to keep things consistent and predictable, values should always be passed around as their designated model/data format. 

In this instance, why isn't the currency/price data already `MarketData`? Is it because it comes from an API that looks different, I presume AlexSDK. When we get external data in to the app, at the earliest possible convenience, we should turn it into our own data models, for interoperability with other code.

With this function, because `MarketData` is created internally, what if I actually have the format we want? Our global app `MarketData` type, it's not compatible with it (because this function just happened to be created when dealing with another format).

I'd say the same if someone wrote a function that accepted `amount` and `symbol`, used `createMoney` then performed some operation on it. It would better accept `Money` directly.

Can't we modify the AlexSDK response and turn this into `MarketData` right away? If we have to `createMarketData` in 5 places to accept it as the input to this function, that's a better imo as we don't deviate from a standard format.

Reply:
I just didn't understand we wanted to convert it so early, that is it. Your request to implement it this way isn't complicated. I converted their `price` to our Money type right away, but didn't realize it would be better to go all-in on `MarketData`. Happy to change it and will keep this in mind moving fwd.

Reply:
@kyranjamie I refactored this so now we select the query response and add the `MarketData` right away for fiat balances. Let me know if this looks good now? I relocated some code from the original swaps work that makes those assets more accessible.